spec/prog/ai/inference_endpoint_replica_nexus_spec.rb

```diff
@@ -132,15 +136,140 @@
   end
 
   describe "#download_lb_cert" do
-    it "downloads lb cert and hops to setup" do
+    it "downloads lb cert and hops to setup_external" do
       expect(sshable).to receive(:cmd).with("sudo inference_endpoint/bin/download-lb-cert")
-      expect { nx.download_lb_cert }.to hop("setup")
+      expect { nx.download_lb_cert }.to hop("setup_external")
+    end
+  end
+
+  describe "#setup_external" do
+    it "hops to setup for vllm" do
+      expect { nx.setup_external }.to hop("setup")
+    end
+
+    it "creates a pod for runpod" do
+      expect(inference_endpoint).to receive(:engine).and_return "runpod"
+      expect(Config).to receive(:operator_ssh_public_keys).and_return "operator ssh key"
+      stub_request(:post, "https://api.runpod.io/graphql")
+        .with(
+          body: "{\"query\":\"query Pods { myself { pods { id name runtime  { ports { ip isIpPublic privatePort publicPort type } } } } }\"}",
+          headers: {
+            "Accept-Encoding" => "deflate, gzip",
+            "Authorization" => "Bearer ",
+            "Content-Type" => "application/json",
+            "Host" => "api.runpod.io"
+          }
+        )
+        .to_return(status: 200, body: "{\"data\": {\"myself\": {\"pods\": []}}}", headers: {})
```

Comment:
nit: make similar to below stub and more readable
```suggestion
        .to_return(status: 200, body: {data: {myself: {pods: []}}}.to_json, headers: {})
```